### PR TITLE
[DUOS-2986] Allow DUOS-000000 in URLs for datasets

### PR DESCRIPTION
**Pull request by fboulnois**
### Addresses

https://broadworkbench.atlassian.net/browse/DUOS-2986

### Summary

Allow DUOS-000000 in URLs for datasets. BONUS: Enables a dataset short link which lives at duos.org/duos-000000

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/libs/ajax.js [hunk 1]
```diff
@@ -356,6 +356,12 @@ export const DataSet = {
     const url = `${await getApiUrl()}/api/dataset/study/${studyId}`;
     return await axios.put(url, studyObject, Config.multiPartOpts());
   },
+
+  getDatasetByDatasetIdentifier: async datasetIdentifier => {
+    const url = `${await getApiUrl()}/api/tdr/${datasetIdentifier}`;
+    const res = await fetchOk(url, Config.authOpts());
```
Comment (rushtong) [inline src/libs/ajax.js]: The fetchOK method will only return if the response is successful. In the case of a 404, for instance, it errors out and the page doesn't display anything. Can we add some logic in `DatasetStatistics` to capture that? You could call `fetchAny` instead and manually handle the non-success case in the UI. Separate from that, we've been trying to migrate to axios (one of our many transitions) so it might be cleaner using that approach.

File: src/pages/DatasetStatistics.jsx [hunk 2]
```diff
@@ -9,15 +9,25 @@ import { formatDate } from '../libs/utils';
...
+  const datasetIdentifier = extractIdentifier(props.match.params);
   const [dataset, setDataset] = useState();
   const [dars, setDars] = useState();
   const [isLoading, setIsLoading] = useState(true);
 
   useEffect(() => {
-    setData(datasetId);
-  }, [datasetId]);
+    DataSet.getDatasetByDatasetIdentifier(datasetIdentifier).then((dataset) => {
+      setData(dataset.dataSetId);
```
Comment (PintoGideon) [inline src/pages/DatasetStatistics.jsx:28]: Fails silently for incorrect dataset identifiers. We could add a notification for an error here for consistency, though I doubt this error state would be triggered as the user navigates to this from the dataset catalog.

**Review by PintoGideon — APPROVED** [review]
Manually tested. Works as expected apart from a minor issue of silent errors.

**Review by rushtong — APPROVED** [review]
Thank you for the changes 👍🏽
